src/flick/flick_auth/controllers/register_controller.py

```diff
@@ -39,15 +41,32 @@ def _create_profile(self, profile_data):
         profile.save()
         return profile
 
+    def _check_token(self, token, type="facebook"):
+        """
+        checks if the token is valid, right now only supports facebook token
+        """
+        if not token:
+            return False
+        # only validate if it is turned on from settings, helps with easier development
+        if settings.VALIDATE_SOCIAL_TOKEN:
+            URL = "https://graph.facebook.com/me"
+            PARAMS = {"access_token": token}
+            response = requests.get(url=URL, params=PARAMS)
+            data = response.json()
+            if data.get("error"):
+                return False
+        return True
+
     def process(self):
         username = self._data.get("username")
         first_name = self._data.get("first_name")
         last_name = self._data.get("last_name")
         profile_pic = self._data.get("profile_pic")
         social_id_token = self._data.get("social_id_token")
         social_id_token_type = self._data.get("social_id_token_type")
-        if not username and not social_id_token:
-            return failure_response("Neither username nor social_id_token supplied.")
+        # check first that we have a valid token:
+        if not self._check_token(social_id_token):
+            return failure_response("social id token is invalid")
```

Comment:
let's try to keep failure response messages consistent: `social_id_token is invalid.`